### ISSUE TITLE
refactor(jans-cedarling): make the tokens optional when calling Cedarling::authorize

### ISSUE THREAD

**Issue by rmarinn** [JanssenProject/jans#10410]
-------------------

### Description

- Development target issue

target issue: #10408

- Implementation Details

Cedarling asks for the `Request` struct when calling `Cedarling::authorize`.

```rs
/// Box to store authorization data
#[derive(Debug, Clone, serde::Deserialize)]
pub struct Request {
    /// Access token raw value
    pub access_token: String,
    /// Id Token raw value
    pub id_token: String,
    /// Userinfo Token raw value
    pub userinfo_token: String,
    /// cedar_policy action
    pub action: String,
    /// cedar_policy resource data
    pub resource: ResourceData,
    /// context to be used in cedar_policy
    pub context: serde_json::Value,
}
```

However, we need to make the `access_token`, `id_token`, and `userinfo_token` optional. Cedarling should make do with what the user gives it then work with that.

-------------------

### Prepare

- [x] Read contribution guidelines
- [x] Read license information

------------------

### Identified code changes

- [ ] make the tokens in the `Request` struct optional

-------------------

### Test cases and code coverage

- [ ] Write unit test to cover added/changed code
- [ ] Update integration tests to cover added/changed code

-------------------

### Document the changes

- [ ] task for updating user guides if needed
- [ ] task for updating installation and configuration guides if needed 
- [ ] task for updating developer documentation if needed
- [ ] task for updating technical documentation if needed

**Comment by olehbozhok** [comment]
According to the @nynymike design we need to create additional structure for tokens
![Image](https://github.com/user-attachments/assets/371b539a-1fff-4f3f-ad6a-631a6249a59f)

**Comment by rmarinn** [comment]
will be implemented with [#10408](https://github.com/JanssenProject/jans/issues/10408).
